src/js/components/Anchor.js

```diff
@@ -3,13 +3,28 @@
 import React, { Children, PropTypes } from 'react';
 import classnames from 'classnames';
 import RightLeftIcon from './icons/LinkRightLeft';
+import iconsMap from '../index-icons';
 
 const CLASS_ROOT = 'anchor';
 
 const Anchor = props => {
   let icon;
 
-  if (props.primary) {
+  if (props.icon && iconsMap.hasOwnProperty(props.icon)) {
+    let CustomIcon = iconsMap[props.icon];
+    icon = <CustomIcon />;
+  } else {
+    icon = <RightLeftIcon />;
```

Comment:
How about logging a warning here instead, indicating that we couldn't find an icon with the supplied name? And we can leave icon undefined for rendering.